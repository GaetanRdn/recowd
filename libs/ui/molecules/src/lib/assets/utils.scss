@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
