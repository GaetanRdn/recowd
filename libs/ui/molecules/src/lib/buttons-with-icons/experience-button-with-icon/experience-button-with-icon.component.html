<a rcButton role="button">
  <rc-material-icon type="outlined">history_edu</rc-material-icon>
  {{ 'experiences' | translate }}
</a>
