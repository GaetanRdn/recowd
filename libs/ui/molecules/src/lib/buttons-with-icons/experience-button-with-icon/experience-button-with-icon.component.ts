import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ButtonDirective, MaterialIconDirective } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-experience-button-with-icon',
  standalone: true,
  imports: [ButtonDirective, MaterialIconDirective],
  templateUrl: './experience-button-with-icon.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ExperienceButtonWithIconComponent {}
