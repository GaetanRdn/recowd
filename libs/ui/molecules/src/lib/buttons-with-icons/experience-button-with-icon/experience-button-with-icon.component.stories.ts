import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { ExperienceButtonWithIconComponent } from './experience-button-with-icon.component';

export default {
  title: 'molecules/buttons-with-icons/experience-button-with-icon',
  decorators: [
    moduleMetadata({
      imports: [ExperienceButtonWithIconComponent],
    }),
  ],
} as Meta<ExperienceButtonWithIconComponent>;

const Template: Story<ExperienceButtonWithIconComponent> = () => ({
  template: `<rc-experience-button-with-icon></rc-experience-button-with-icon>`,
});

export const Default = Template.bind({});
