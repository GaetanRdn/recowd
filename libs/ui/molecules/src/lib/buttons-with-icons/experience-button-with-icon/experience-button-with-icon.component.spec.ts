import { TestBed, waitForAsync } from '@angular/core/testing';
import { Component } from '@angular/core';
import { TemplateLookup } from '@recowd/test/utils';
import { ExperienceButtonWithIconComponent } from './experience-button-with-icon.component';

describe('ExperienceButtonWithIconComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [ExperienceButtonWithIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());
});

@Component({
  template: ` <rc-experience-button-with-icon></rc-experience-button-with-icon>`,
})
export class HostComponent {}
