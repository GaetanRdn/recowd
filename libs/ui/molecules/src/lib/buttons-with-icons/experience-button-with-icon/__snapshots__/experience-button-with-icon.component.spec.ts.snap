// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceButtonWithIconComponent should create 1`] = `
<rc-experience-button-with-icon>
  <a
    class="rc-button"
    rcbutton=""
  >
    <rc-material-icon
      class="material-icons-outlined"
      type="outlined"
    >
      history_edu
    </rc-material-icon>
     Exp√©riences

  </a>
</rc-experience-button-with-icon>
`;
